Survey/Survey/Sources/Presentation/Modules/Entry/EntryView.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  EntryView.swift
+//  Survey
+//
+//  Created by Taher on 25/10/22.
+//
+
+import SwiftUI
+
+struct EntryView: View {
+
+    @ObservedObject var viewModel: EntryViewModel
+    @State var logoOpacity = 0.0
+
+    var body: some View {
+        switch viewModel.viewType {
+        case .splash:
+            SplashView(logoOpacity: $logoOpacity)
+                .onAnimationCompleted(for: logoOpacity) {
+                    viewModel.didFinishSplashAnimation.send()
+                }
+        case .login:
+            LoginView()
+        }
+    }
```

Comment:
As my concern about the necessary of `EntryViewType`, I'm curious why we need this `EntryView` when we can put this directly to the `AppCoordinator`. Kind ly provide more insights for this 🙏